src/main/java/seedu/address/storage/XmlAdaptedTask.java

```diff
@@ -52,7 +52,7 @@ public XmlAdaptedTask(ReadOnlyTask source) {
         priority = source.getPriority()
                 .map(Priority::getValue)
                 .map(Priority.Type::name)
-                .orElse(Priority.Type.NONE.name());
+                .orElse(null);
```

Comment:
Changing the `Optional` type to `nullable` would mean performing a simple `.get()` would suffice. `.orElse(null)` would serve no purpose here.